maths/numerical_analysis/lagrange_interpolation.py

```diff
@@ -0,0 +1,60 @@
+def lagrange_interpolation(x: list[float], y: list[float], value: float) -> float:
+    """
+    Calculate the Lagrange interpolation of a function based on provided data points.
+
+    Parameters:
+        x (list[float]): List of x-values (independent variable).
+        y (list[float]): List of corresponding y-values (dependent variable).
+        value (float): The value at which to interpolate the function.
+
+    Returns:
+        float: The interpolated value of the function at the specified 'value'.
+
+    Examples:
+        >>> x = [5.0, 6.0, 9.0, 11.0]
+        >>> y = [12.0, 13.0, 14.0, 16.0]
+        >>> lagrange_interpolation(x, y, 10.0)
+        14.666666666666666
```

Comment:
Please add tests that include empty lists, zeros, negative numbers, real floats (5.5), ints instead of floats (5), and str("hello", "world) and tuples x = (5, 0, 9.9, -11.0), y = [-12.0, 13.13, 0, 16]